lib/semian/shared_memory_object.rb

```diff
@@ -0,0 +1,50 @@
+module Semian
+  class SharedMemoryObject
+
+    @@type_size = {}
+    def self.sizeof(type)
+      size = (@@type_size[type.to_sym] ||= (respond_to?(:_sizeof) ? _sizeof(type.to_sym) : 0))
+      raise TypeError.new("#{type} is not a valid C type") if size <=0
+      size
+    end
+
+    def semid
+      -1
+    end
+
+    def shmid
+      -1
+    end
+
+    def execute_atomically
+      yield if block_given?
+    end
+
+    def shared?
+      @using_shared_memory ||= Semian.extension_loaded && @using_shared_memory
+    end
+
+    def destroy
+      _destroy if respond_to?(:_destroy) && @using_shared_memory
+    end
+
+    # Functions dealing with acquiring memory
```

Comment:
I don't think we need a "function dealing with acquiring memory" comment on a function that's called "acquire_memory"